kong/db/strategies/postgres/connector.lua

```diff
@@ -158,71 +142,76 @@ function _mt:setkeepalive()
   end
 
   if not ok then
+    log(WARN, err)
     return nil, err
   end
 
   return true
 end
 
 
-function _mt:query(sql)
-  local connection = self.connection
-
-  if connection then
-    return connection:query(sql)
+local function query(sql, self)
+  if self.connection and self.connection.sock then
```

Comment:
Wouldn't the fix be more concise if it simply added checks for `self.connection.sock` in the places you just added it? I'm not quite sure making `setkeepalive()` and `connect()` local functions really helps? It seems like it just makes the logic more difficult to follow: we sometimes call `setkeepalive()`, and some other times call `self:setkeepalive()` - the latter just unsetting `self.connection`, but that seems unnecessary since the former will already unset `self.connection.sock`, and we now check for this value in `query()`?)